### PR TITLE
fulfil the newest microstorage interface

### PR DESCRIPTION
**Pull request by kopiczko**
- remove Create method
- support K & KV
- detect minikube IP in integration test

### CODE REVIEW

File: storage_integration_test.go [hunk 1]
```diff
@@ -49,7 +51,16 @@ func init() {
...
+	var serverDefault string
+	{
+		out, err := exec.Command("minikube", "ip").Output()
+		if err == nil {
+			minikubeIP := strings.TrimSpace(string(out))
+			serverDefault = "https://" + string(minikubeIP) + ":8443"
+		}
+	}
+
+	flag.StringVar(&server, "integration.server", serverDefault, "Kubernetes API server address")
```
Comment (kopiczko) [inline storage_integration_test.go]: ninja fix: detect minikube IP

File: storage.go [hunk 2]
```diff
@@ -216,28 +216,8 @@ func (s *Storage) Boot(ctx context.Context) error {
...
-	if err != nil {
-		return microerror.Mask(err)
-	}
-
-	err = s.Put(ctx, key, value)
-	if err != nil {
-		microerror.Mask(err)
-	}
-	return nil
-}
```
Comment (kopiczko) [inline storage.go]: Create is removed from the interface.

File: storage.go [hunk 3]
```diff
@@ -216,28 +216,8 @@ func (s *Storage) Boot(ctx context.Context) error {
...
-
-func (s *Storage) Put(ctx context.Context, key, value string) error {
-	var err error
-
-	key, err = microstorage.SanitizeKey(key)
-	if err != nil {
-		return microerror.Mask(err)
-	}
+func (s *Storage) Put(ctx context.Context, kv microstorage.KV) error {
+	key, value := kv.Key(), kv.Val()
```
Comment (kopiczko) [inline storage.go]: We don't need sanitization anymore. KV and K are immutable and validated upon creation. This is also reflected in the test suite.

Comment (MarcelMue) [reply storage.go]: I really like this 👍

**Pull request merged**